### ISSUE TITLE
Validator: -proxy Flag treated as File to validate

### ISSUE THREAD

**Issue by alexanderkiel** [hapifhir/org.hl7.fhir.core#145]
If I run the validator CLI version 4.2.4-SNAPSHOT with a `-proxy` flag, it tries to open `-proxy` as a file:

```
$ java -jar validator.jar -proxy <host>:<port> -version 4.0.1 -ig <ig.pack> <files-to-validate>
FHIR Validation tool Version 4.2.4-SNAPSHOT (Git# a3d7acdf351d). Built 2020-02-28T15:17:35.66+11:00 (3 days old)
Detected Java version: 11.0.3 from /Library/Java/JavaVirtualMachines/jdk-11.0.3.jdk/Contents/Home on x86_64 (64bit). 4096MB available
Arguments: -proxy <host>:<port> -version 4.0.1 -ig <ig.pack> <files-to-validate>
Directories: Current = ..., Package Cache = ...
  .. FHIR Version 4.0, definitions from hl7.fhir.r4.core#4.0.1
  .. connect to tx server @ http://tx.fhir.org
Installing hl7.fhir.r4.core#4.0.1 to the package cache
  Fetching:....................................................................................................
  Installing: .................................................................................................... done.
Installing hl7.fhir.xver-extensions#0.0.1 to the package cache
  Fetching:
  Installing:  done.
    (v4.0.1)
+  .. load IG from <ig.pack>
  .. validate [-proxy <host>:<port>, <files-to-validate>]
Exception in thread "main" java.io.IOException: File -proxy does not exist
        at org.hl7.fhir.validation.ValidationEngine.handleSource(ValidationEngine.java:1048)
        at org.hl7.fhir.validation.ValidationEngine.handleSources(ValidationEngine.java:1023)
        at org.hl7.fhir.validation.ValidationEngine.validate(ValidationEngine.java:983)
        at org.hl7.fhir.validation.Validator.main(Validator.java:659)
```

I replaced the real paths with placeholders like `<...>`.

**Comment by patrick-werner** [comment]
in my experience the first argument of the validator has to be the file/folder you want to validate, the arguments have to follow after it.
Does this fix your issue?

**Comment by grahamegrieve** [comment]
It shouldn't be dependent on order.

**Comment by alexanderkiel** [comment]
@patrick-werner No, unfortunately that doesn't fix my issue.

**Comment by alexanderkiel** [comment]
I think adding `-proxy` somewhere in the if-statement starting [here](https://github.com/hapifhir/org.hl7.fhir.core/blob/9178945241ecad08918fef38a15b89710e692da6/org.hl7.fhir.validation/src/main/java/org/hl7/fhir/validation/Validator.java#L418) should solve the issue.

**Comment by patrick-werner** [comment]
> It shouldn't be dependent on order.

oh indeed. I remembered it wrong (maybe it was the case in a old version).
Sorry for the confusion.

**Comment by grahamegrieve** [comment]
fixed

**Comment by alexanderkiel** [comment]
Fixed in ca240e4
